src/pages/grants/grantsPage/sections/CommunityVoting.tsx

```diff
@@ -49,152 +31,64 @@ export const CommunityVoting = ({
   title,
   isClosed,
   isCompetitionVote,
+  votingSystem,
 }: Props) => {
   const { t } = useTranslation()
-  const isMobile = useMobileMode()
-  const classes = useStyles()
-  const modalProps = useProjectFundingModal()
+  const fundingModalProps = useProjectFundingModal()
+  const { onOpen, onClose, isOpen } = useDisclosure()
+  const { user, isLoggedIn } = useAuthContext()
 
   if (!applicants) {
     return null
   }
 
-  const canVote = grantHasVoting && grantStatus === GrantStatusEnum.FundingOpen
-
-  const renderWidgetItem = (funding: GrantApplicantFunding, contributorsCount: number) => {
-    return (
-      <HStack gap={5}>
-        {isCompetitionVote && (
-          <Box display={'flex'} alignItems="center" flexDirection={'column'}>
-            <Box display={'flex'} alignItems="center">
-              <Text fontWeight={'700'} fontSize={'26px'} fontFamily={fonts.livvic} color="primary.500">
-                {contributorsCount || 0}
-              </Text>
-            </Box>
+  let sortedApplicants = applicants
 
-            <Text fontWeight={'400'} fontSize="10px" fontStyle="normal" color="neutral.900">
-              {t('voters')}
-            </Text>
-          </Box>
-        )}
-        <WidgetItem subtitle={!isClosed ? t('worth of votes') : t('distributed')}>
-          {getShortAmountLabel(
-            !isClosed ? funding.communityFunding : funding.grantAmount + funding.communityFunding || 0,
-          )}
-        </WidgetItem>
-      </HStack>
-    )
+  if (user) {
```

Comment:
Do I understand correctly that this is for putting the applicants the logged in user contributed to at the top? And they are not sorted by contribution amount right?

If that's the case, I think sorting is not the most efficient way to do this, because you're iterating over each applicant contributors twice. Instead you can loop over the applicants once and check if the userId is contained in the contributors. if it is add it to one list, if it isn't add it to another, then merge the two lists together. You can also sort the first list by amount contributed before merging as a bonus.

Or, use a single array, and push the applicant in the frontend of it if the user has contributed to it (eg with `unshift`), and otherwise push it in the back of the array.